attest/attest.go

```diff
@@ -20,6 +20,28 @@ import (
 	"github.com/veraison/go-cose"
 )
 
+type Params struct {
```

Comment:
If `Params` and all of its members are exported, what's the purpose of the functional options?

Reply:
Could make them private. Wasn't sure how this pattern usually looks.

Reply:
It's a good pattern! Typically you'd use it on a constructor, so maybe something like [this](https://go.dev/play/p/MN-orWMzGzq). One of the advantages of operating directly on a struct that does the work is that you have one less layer of plumbing you have to maintain when adding or removing functionality.

Reply:
Nice something like that makes sense. 

In the example, were you thinking Verifier struct would do all the attestation work or just the verifying certificate work? I think it makes sense if one struct was doing all the work because it would make it easier for the consumer to configure.

Reply:
Updated now let me know what you think.